src/analyses/variable-sensitivity/variable_sensitivity_domain.cpp

```diff
@@ -0,0 +1,345 @@
+
+\*******************************************************************/
+
+void variable_sensitivity_domaint::transform(
+    locationt from,
+    locationt to,
+    ai_baset &ai,
+    const namespacet &ns)
+{
+  #ifdef DEBUG
+  std::cout << "Transform from/to:\n";
+  std::cout << from->location_number << " --> "
+            << to->location_number << std::endl;
+  #endif
+
+  const goto_programt::instructiont &instruction=*from;
+  switch(instruction.type)
+  {
+  case DECL:
+    // Creates a new variable, which should be top
+    // but we don't store top so ... no action required
+    break;
+
+  case DEAD:
+    {
+    // Assign to top is the same as removing
+    abstract_object_pointert top_object=
+      abstract_state.abstract_object_factory(
+        to_code_dead(instruction.code).symbol().type(), ns, true);
+    abstract_state.assign(
+      to_code_dead(instruction.code).symbol(), top_object, ns);
+    }
+    break;
+
+  case ASSIGN:
+    {
+      const code_assignt &inst = to_code_assign(instruction.code);
+
+      // TODO : check return values
+      abstract_object_pointert r = abstract_state.eval(inst.rhs(), ns);
+      abstract_state.assign(inst.lhs(), r, ns);
+    }
+    break;
+
+  case GOTO:
+    {
+      if (1) // (flow_sensitivity == FLOW_SENSITIVE)
+      {
+        // Get the next line
+        locationt next=from;
+        next++;
+        // Is this a GOTO to the next line (i.e. pointless)
+        if(next!=from->get_target())
+        {
+          if(to==from->get_target())
+          {
+            // The AI is exploring the branch where the jump is taken
+            abstract_state.assume(instruction.guard, ns);
+          }
+          else
+          {
+            // Exploring the path where the jump is not taken - therefore assume
+            // the condition is false
+            abstract_state.assume(not_exprt(instruction.guard), ns);
+          }
+        }
+        // ignore jumps to the next line, we can assume nothing
+      }
+    }
+    break;
+
+  case ASSUME:
+    abstract_state.assume(instruction.guard, ns);
+    break;
+
+  case FUNCTION_CALL:
+    // FIXME : Ignore as not yet interprocedural
+    break;
+
+  case END_FUNCTION:
+    // FIXME : Ignore as not yet interprocedural
+    break;
+
+    /***************************************************************/
+
+  case ASSERT:
+    // Conditions on the program, do not alter the data or information
+    // flow and thus can be ignored.
+    // Checking of assertions can only be reasonably done once the fix-point
+    // has been computed, i.e. after all of the calls to transform.
+    break;
+
+  case SKIP:
+  case LOCATION:
+    // Can ignore
+    break;
+
+  case RETURN:
+    throw "return instructions are depreciated";
+
+  case START_THREAD:
+  case END_THREAD:
+  case ATOMIC_BEGIN:
+  case ATOMIC_END:
+    throw "threading not supported";
+
+  case THROW:
+  case CATCH:
+    throw "exceptions not handled";
+
+  case OTHER:
+//    throw "other";
```

Comment:
@martin-cs I had to comment this out early on as there are other instructions inside the CPROVER_Initailize I think - I suppose this is OK?